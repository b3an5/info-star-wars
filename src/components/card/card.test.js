import React from 'react';
import Card from './card.js';
import { shallow } from 'enzyme';

describe('card', () => {

  let wrapper 
  let mockCategoryNameP = 'People'
  let mockCategoryNameV = 'Vehicles'
  let mockCategoryNamePl = 'Planets'
  let mockCategoryV = {
    name: 'theo',
    model: 'monkey',
    vehicle_class: 'earth',
    passengers: 1
  }
  let mockCategoryP = {
    name: 'theo',
    species: 'monkey',
    planet: 'earth',
    population: 1,
    url: 'www.mockdatatheo.com'
  }
  let mockCategoryPl = {
    name: 'theo',
    terrain: 'monkey',
    climate: 'earth',
    population: 100,
    residents: ['a', 'b']
  }


  beforeEach(() => {
    wrapper = shallow(
      <Card
        category={mockCategoryP}
        categoryName={mockCategoryNameP}
        key={mockCategoryP.url} 
      />
    )
  })

  it('should match snapshot of people', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot of vehicles', () => {
    wrapper = shallow(
      <Card
        category={mockCategoryV}
        categoryName={mockCategoryNameV}
        key={mockCategoryV.url}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot of planets', () => {
    wrapper = shallow(
      <Card
        category={mockCategoryPl}
        categoryName={mockCategoryNamePl}
        key={mockCategoryPl.url}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})