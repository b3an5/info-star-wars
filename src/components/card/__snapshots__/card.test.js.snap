// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match snapshot of people 1`] = `ShallowWrapper {}`;

exports[`card should match snapshot of planets 1`] = `ShallowWrapper {}`;

exports[`card should match snapshot of vehicles 1`] = `ShallowWrapper {}`;
